Cargo.toml

```diff
@@ -20,6 +20,7 @@ exclude = [
 ]
 
 [dependencies]
+byteorder = "1.5.0"
```

Comment:
why are we adding an extra dependency, can we instead rely on `u32::to_be_bytes`? https://doc.rust-lang.org/std/primitive.u32.html#method.to_be_bytes